### ISSUE TITLE
I am curious about the implement of crf_accuracy

### ISSUE THREAD

**Issue by logicthinkedu** [xuxingya/tf2crf#12]
Hello Sir,

I am curious about the implement of crf_accuracy
I builded a NER model and get a very high testing accuracy.

But when I use the Model to recognize entity , there still many Error.
I am curious about the implement of crf_accuracy?  Is it the same with tensorflow.contrib.crf?

**Comment by xuxingya** [comment]
Yes, it is the same with tensorflow.contrib.crf. Is your loss decreasing?
If so it won't be the problem of model, but overfitting or imbalance data.
Usually I won't evaluate a NER model simply by accuracy, because it usually
multiple tags and imbalanced data.
I recommend you to evaluate the confusion matrix of your predicted result.
You can use f1_score and classification_report from sklearn.metrics.

**Comment by logicthinkedu** [comment]
thanks for your quickly reply

I have a high accuracy, but the loss is still high.
I will check my code

**Comment by xuxingya** [comment]
you can use Callbacks if you want to use the f1 score. I post my code as an example:
```
class Evaluate(Callback):
    def __init__(self, validation_data, version):
        super(Evaluate, self).__init__()
        self.F1 = []
        self.best = 0
        self.validation_data = validation_data
        self.version = version

    def on_epoch_end(self, epoch, logs=None):
        f1, p, r = self.evaluate()
        self.F1.append(f1)
        if f1 > self.best:
            self.best = f1
            export_path = os.path.join(train_config.MODEL_PATH, 'block_model', str(self.version))
            tf.keras.models.save_model(
                model,
                export_path,
                overwrite=True,
                include_optimizer=True,
                save_format=None,
                signatures=None,
                options=None
            )
        print('f1: %.4f, best f1: %.4f, precision: %.4f, recall: %.4f\n' % (f1, self.best, p, r))

    def evaluate(self):
        val_true = self.validation_data[1].flatten()
        val_pred = self.model.predict(self.validation_data[0], 128)
        val_pred = val_pred.flatten()
        y_true, y_pred = zip(*filter(lambda x: x[0]!=0, zip(val_true, val_pred)))
        return f1_score(y_true, y_pred, average='macro'), precision_score(y_true, y_pred,
                                                                          average='macro'), recall_score(y_true, y_pred,

                                                                                                         average='macro')
```
